### PR TITLE
Fix unhandled rejection "_signTypedData"

### PR DESCRIPTION
**Pull request by samuveth**
Fix https://discord.com/channels/707079246388133940/747049240022482974/917090820295909419

### CODE REVIEW

File: src/sign/index.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default class Client {
     if (!message.timestamp)
       message.timestamp = parseInt((Date.now() / 1e3).toFixed());
     const data: any = { domain, types, message };
-    const sig = await signer._signTypedData(domain, data.types, message);
+    const sig = await signer?._signTypedData(domain, data.types, message);
```
Comment (ChaituVR) [inline src/sign/index.ts:56]: Why do we get signer undefined? 🤔 what should happen if it is undefined? 
If it is like this we send empty sig to the hub, right? maybe would be good to show some error message?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/sign/index.ts (hunk 1 above)
Comment (samuveth) [reply src/sign/index.ts:56]: I can't reproduce signer undefined. Anytime I log the signer it's available. Not sure why it's showing this error.

Comment (samuveth) [reply src/sign/index.ts:56]: Ok can reproduce now.

File: src/sign/index.ts [hunk 2]
```diff
@@ -47,8 +47,9 @@ export default class Client {
 
   async sign(web3: Web3Provider | Wallet, address: string, message, types) {
     let signer;
-    if (web3 instanceof Wallet) signer = web3;
-    if (web3 instanceof Web3Provider) signer = web3.getSigner();
+    // @ts-ignore
+    if (web3?.getSigner) signer = web3.getSigner();
+    else signer = web3;
```
Comment (bonustrack) [inline src/sign/index.ts:52]: Could be simplified to
```suggestion
  const signer = web3?.getSigner ? web3.getSigner() : web3;
```

**Review by bonustrack — APPROVED** [review]
Looks good!

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**